job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java

```diff
@@ -174,37 +172,26 @@ private boolean shouldIgnore(URIStatus status) {
   @Override
   public SerializableVoid runTask(CompactConfig config, ArrayList<CompactTask> tasks,
       RunTaskContext context) throws Exception {
-    Closer closer = Closer.create();
-    boolean closed = false;
-    Compactor compactor = new SequentialCompactor();
     for (CompactTask task : tasks) {
       ArrayList<String> inputs = task.getInputs();
       if (inputs.isEmpty()) {
         continue;
       }
       AlluxioURI output = new AlluxioURI(task.getOutput());
-      List<TableReader> readers = Lists.newArrayList();
-      TableWriter writer = null;
-      try {
+
+      TableSchema schema;
+      try (TableReader reader = TableReader.create(new AlluxioURI(inputs.get(0)), config.getPartitionInfo())) {
+        schema = reader.getSchema();
+      }
+
+      try (TableWriter writer = TableWriter.create(schema, output)) {
         for (String input : inputs) {
-          readers.add(closer.register(TableReader.create(new AlluxioURI(input),
-              config.getPartitionInfo())));
-        }
-        TableSchema schema = readers.get(0).getSchema();
-        writer = closer.register(TableWriter.create(schema, output));
-        compactor.compact(readers, writer);
-      } catch (Throwable t) {
-        closer.close();
-        closed = true;
-        try {
-          context.getFileSystem().delete(output); // output is the compacted file
```

Comment:
Shouldn't we delete the output if there is a failure?

Reply:
Sorry. I just needed something quick to retrigger build. I went back to deleting it again.